cmake/modules/BuildArrow.cmake

```diff
@@ -0,0 +1,85 @@
+# copied from BuildRocksDB.cmake
+
+function(build_arrow)
+  # only enable the parquet component
+  set(arrow_CMAKE_ARGS -DARROW_PARQUET=ON)
+  list(APPEND arrow_CMAKE_ARGS -DPARQUET_MINIMAL_DEPENDENCY=ON)
+
+  # only use preinstalled dependencies for arrow, don't fetch/build any
+  list(APPEND arrow_CMAKE_ARGS -DARROW_DEPENDENCY_SOURCE=SYSTEM)
+
+  # only build static library
+  list(APPEND arrow_CMAKE_ARGS -DARROW_BUILD_SHARED=OFF)
+  list(APPEND arrow_CMAKE_ARGS -DARROW_BUILD_STATIC=ON)
```

Comment:
yes, more broadly, what is built statically, with this option?  does it leave the arrow and parquet targets as static libs?  if so, yeah, I am skeptical.

Reply:
i just wanted to avoid packaging those shared libraries with radosgw. it seems like that would lead to conflicts if/when distros provide them. is shared linkage worth this risk?

Reply:
> yes, more broadly, what is built statically, with this option? does it leave the arrow and parquet targets as static libs? if so, yeah, I am skeptical.

the intent was to build arrow and parquet statically, but leave their dependencies as shared. we'll just need to make sure that arrow's cmake finds the same libraries ceph is using, which might be tricky if ceph is building them from submodules

Reply:
ok, I'd like to optimize later, but do whatever is cleanest for now